src/transformStateWithClones.js

```diff
@@ -6,8 +6,65 @@
  *
  * @return {Object[]}
  */
+// function transformStateWithClones(state, actions) {
+//   const stateCopy = { ...state };
+
+//   const actionsResult = [];
+
+//   for (const ch of actions) {
+//     if (ch.type === 'addProperties') {
+//       Object.assign(stateCopy, ch.extraData);
+//       actionsResult.push(Object.assign({}, stateCopy));
+//     };
+
+//     if (ch.type === 'removeProperties') {
+//       for (const cn of ch.keysToRemove) {
+//         delete (stateCopy[cn]);
+//       };
+//       actionsResult.push(Object.assign({}, stateCopy));
+//     }
+
+//     if (ch.type === 'clear') {
+//       for (const cm in stateCopy) {
+//         delete (stateCopy[cm]);
+//       };
+//       actionsResult.push(Object.assign({}, stateCopy));
+//     };
+//   };
+
+//   return actionsResult;
+// }
+
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let stateCopy = { ...state };
+
+  const actionsResult = [];
+
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clear = 'clear';
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use a destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, ....... } = action;
    
    switch (type) {
```